packages/create-pantheon-decoupled-kit/src/index.ts

```diff
@@ -0,0 +1,135 @@
+import chalk from 'chalk';
+import inquirer from 'inquirer';
+import nodePlop, { NodePlopAPI } from 'node-plop';
+import minimist from 'minimist';
+import type { Answers, QuestionCollection } from 'inquirer';
+import type { ParsedArgs, Opts as MinimistOptions } from 'minimist';
+import type { DecoupledKitGenerator } from '@cli/src/types';
+
+const __filename = new URL('.', import.meta.url).pathname;
+
+/**
+ * Set generator based on exports from src/generators
+ * @param generators An array of plop Generators with an added name field. @see {@link DecoupledKitGenerator}.
+ * @returns {Promise<NodePlopAPI>} plop
+ */
+export const setGenerators = async (
+	generators: DecoupledKitGenerator[],
+): Promise<NodePlopAPI> => {
+	const plop = await nodePlop();
+	for (const generator of Object.values(generators)) {
+		plop.setGenerator(generator.name, generator);
+	}
+	return plop;
+};
+
+/**
+ *  Parses CLI arguments using `minimist`
+ * @see {@link https://www.npmjs.com/package/minimist#var-argv--parseargsargs-opts}
+ * @param cliArgs - an array of strings.
+ * @defaultValue `process.argv.slice(2)`
+ * @returns minimist parsed args
+ */
+export const parseArgs = (
+	cliArgs: string[] = process.argv.slice(2),
+): ParsedArgs => {
+	// parse any command line arguments passed into the create command
+	// to pass to the generator prompts and skip them.
+	// useful for CI and testing purposes
+	const options: MinimistOptions = {
+		// these options tell minimist which --args are
+		// booleans and which are strings.
+		// we will probably want these soon
+		boolean: ['force', 'silent' /**noInstall */],
+		// might want these too
+		//string: ['appName', 'directory'],
+	};
+	const args: ParsedArgs = minimist(cliArgs, options);
+
+	return args;
+};
+
+/**
+ * Initializes the CLI prompts based on parsed arguments
+ * @param args - {@link typeof ParsedArgs}
+ * @param DecoupledKitGenerators - An array of plop Generators with an added name field. @see {@link DecoupledKitGenerator}.
+ * @remarks positional args are assumed to be generator names. Multiple generators can be queued up this way. Any number of prompts may be skipped by passing in the prompt name via flag.
+ * @returns Promise<void>
```

Comment:
Running the docs I'm getting an error. Looks like it is treating `<void>` here as JSX :(

```
Compiled with problems:X

ERROR in ./docs/Packages/create-pantheon-decoupled-kit/modules.md

Module build failed (from ../node_modules/.pnpm/@docusaurus+mdx-loader@2.2.0_zneentkx4scexj4pzosurqq55y/node_modules/@docusaurus/mdx-loader/lib/index.js):
SyntaxError: /Users/brianperry/repos/decoupled-kit-js/web/docs/Packages/create-pantheon-decoupled-kit/modules.md: Expected corresponding JSX closing tag for <void>. (40:20)

  38 | <h4 {...{"id":"returns"}}>{`Returns`}</h4>
  39 | <p><inlineCode parentName="p">{`Promise`}</inlineCode>{`<`}<inlineCode parentName="p">{`void`}</inlineCode>{`>`}</p>
> 40 | <p>{`Promise`}<void></p>
     |                     ^
  41 | <h4 {...{"id":"defined-in"}}>{`Defined in`}</h4>
  42 | <p><a parentName="p" {...{"href":"https://github.com/backlineint/decoupled-kit-js/blob/0c623b70/packages/create-pantheon-decoupled-kit/src/index.ts#L59"}}>{`index.ts:59`}</a></p>
  43 | <hr></hr>
```

Reply:
That is very strange given some of our other docs have the same or a similar signature. Digging into it!